pkg/collector/upgrade/v0_57_0.go

```diff
@@ -0,0 +1,74 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package upgrade
+
+import (
+	"fmt"
+	"strings"
+
+	"gopkg.in/yaml.v2"
+
+	"github.com/open-telemetry/opentelemetry-operator/apis/v1alpha1"
+	"github.com/open-telemetry/opentelemetry-operator/pkg/collector/adapters"
+
+	corev1 "k8s.io/api/core/v1"
+)
+
+func upgrade0_57_0(u VersionUpgrade, otelcol *v1alpha1.OpenTelemetryCollector) (*v1alpha1.OpenTelemetryCollector, error) {
+
+	if len(otelcol.Spec.Config) == 0 {
+		return otelcol, nil
+	}
+
+	otelCfg, err := adapters.ConfigFromString(otelcol.Spec.Config)
+	if err != nil {
+		return otelcol, fmt.Errorf("couldn't upgrade to v0.57.2, failed to parse configuration: %w", err)
+	}
+
+	//Remove externsions.healthcheckextension - Remove deprecated port field from config. (#12668)
+	extensionsConfig, ok := otelCfg["extensions"].(map[interface{}]interface{})
+	if !ok {
+		// In case there is no health check config.
+		return otelcol, nil
+	}
+
+	for keyExt, valExt := range extensionsConfig {
+		if strings.HasPrefix(keyExt.(string), "health_check") {
+			switch extensions := valExt.(type) {
+			case map[interface{}]interface{}:
+				if port, ok := extensions["port"]; ok {
+					endpointV := extensions["endpoint"]
+					extensions["endpoint"] = fmt.Sprintf("%s:%s", endpointV, port)
+					delete(extensions, "port")
+
+					otelCfg["extensions"] = extensionsConfig
+					res, err := yaml.Marshal(otelCfg)
+					if err != nil {
+						return otelcol, fmt.Errorf("couldn't upgrade to v0.57.2, failed to marshall back configuration: %w", err)
+					}
+
+					otelcol.Spec.Config = string(res)
+					//trings.ReplaceAll(string(res), " null", "")
+					existing := &corev1.ConfigMap{}
+					updated := existing.DeepCopy()
```

Comment:
can you explain lines 64 and 65? Why are you creating a config map, then creating a copy and then using the copy as an object to create an event on? 

Why are we not creating the event on the otelcol CR?

Reply:
Right, it's better to post the otelcol. The first idea was to replicate a configmap.